### PR TITLE
[MRG] Fix axis label missing bug

### PR DESCRIPTION
**Pull request by chenghuzi**
In the past, it was common for the y-axis to disappear after editing or creating a figure. A protection mechanism is added now to prevent such bugs.

image

### CODE REVIEW

File: hnn_core/gui/_viz_manager.py [hunk 1]
```diff
@@ -97,13 +97,20 @@ def _update_ax(fig, ax, single_simulation, sim_name, plot_type, plot_config):
     net_copied = copy.deepcopy(single_simulation['net'])
 
     for dpl in dpls_copied:
-        dpl.smooth(plot_config['dipole_smooth']).scale(
-            plot_config['dipole_scaling'])
+        if plot_config['dipole_smooth'] > 0:
+            dpl.smooth(plot_config['dipole_smooth']).scale(
+                plot_config['dipole_scaling'])
+        else:
+            dpl.scale(plot_config['dipole_scaling'])
```
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:104]:
```suggestion
        if plot_config['dipole_smooth'] > 0:
            dpl.smooth(plot_config['dipole_smooth'])
        dpl.scale(plot_config['dipole_scaling'])
```

more compact/clear?

**Comment by jasmainak** [comment]
@chenghuzi can you set PR to MRG when it's good to go for you?

**Comment by chenghuzi** [comment]
> @chenghuzi can you set PR to MRG when it's good to go for you?

I think this is already MRG, though the annoying link check error is still there.

**Comment by chenghuzi** [comment]
> > @chenghuzi can you set PR to MRG when it's good to go for you?
> 
> I think this is already MRG, though the annoying link check error is still there.

alright, now we have two more ignored links...

**Comment by rythorpe** [comment]
> > > @chenghuzi can you set PR to MRG when it's good to go for you?
> > 
> > 
> > I think this is already MRG, though the annoying link check error is still there.
> 
> alright, now we have two more ignored links...

Out of curiosity, is this the right way to fix the linkcheck error? I've tried accessing both of those links on numerous occasions, and they work every time. Is there a way to allow the linkcheck to accept url redirects?

**Comment by chenghuzi** [comment]
> > > > @chenghuzi can you set PR to MRG when it's good to go for you?
> > > 
> > > 
> > > I think this is already MRG, though the annoying link check error is still there.
> > 
> > 
> > alright, now we have two more ignored links...
> 
> Out of curiosity, is this the right way to fix the linkcheck error? I've tried accessing both of those links on numerous occasions, and they work every time. Is there a way to allow the linkcheck to accept url redirects?

I've tried with browsers and they look fine. I don't think this is a problem of redirection as Sphinx can handle that without errors with some of other links. It might be the strategy used by JNP as I saw some people claimed that using a more realistic user-agent may solve the issue.

**Comment by chenghuzi** [comment]
> > > > > @chenghuzi can you set PR to MRG when it's good to go for you?
> > > > 
> > > > 
> > > > I think this is already MRG, though the annoying link check error is still there.
> > > 
> > > 
> > > alright, now we have two more ignored links...
> > 
> > 
> > Out of curiosity, is this the right way to fix the linkcheck error? I've tried accessing both of those links on numerous occasions, and they work every time. Is there a way to allow the linkcheck to accept url redirects?
> 
> I've tried with browsers and they look fine. I don't think this is a problem of redirection as Sphinx can handle that without errors with some of other links. It might be the strategy used by JNP as I saw some people claimed that using a more realistic user-agent may solve the issue.

check this one:

image
it's indeed JNP

**Comment by rythorpe** [comment]
Interesting. So it's just an issue on JNP's side that we can safely ignore for our linkcheck?

**Comment by chenghuzi** [comment]
> Interesting. So it's just an issue on JNP's side that we can safely ignore for our linkcheck?

Not sure if it's safe but that might be the only way to get rid of those errors...

**Comment by jasmainak** [comment]
This PR looks good pending the CircleCI fix which we should do first

**Comment by rythorpe** [comment]
With #603 merged, I think it's fair to trust that CircleCI will pass here as well. I'm going to rebase and merge to maintain momentum.

**Comment by rythorpe** [comment]
Thanks @chenghuzi!

**Pull request merged**